core/controlplane/config/templates/cloud-config-worker

```diff
@@ -693,6 +659,74 @@ write_files:
       rkt rm --uuid-file=/var/run/coreos/set-aws-environment.uuid || :
 {{end}}
 
+  {{if .Experimental.AwsNodeLabels.Enabled -}}
+  - path: /opt/bin/kube-node-label
+    permissions: 0700
+    owner: root:root
+    content: |
+      #!/bin/bash -e
+      set -ue
+
+      INSTANCE_ID="$(/usr/bin/curl -s http://169.254.169.254/latest/meta-data/instance-id)"
+      SECURITY_GROUPS="$(/usr/bin/curl -s http://169.254.169.254/latest/meta-data/security-groups | tr '\n' ',')"
+      {{if not .SpotFleet.Enabled -}}
+      AUTOSCALINGGROUP="$(/usr/bin/docker run --rm --net=host \
+        {{.AWSCliImage.RepoWithTag}} aws \
+        autoscaling describe-auto-scaling-instances \
+        --instance-ids ${INSTANCE_ID} --region {{.Region}} \
+        --query 'AutoScalingInstances[].AutoScalingGroupName' --output text)"
+      LAUNCHCONFIGURATION="$(/usr/bin/docker run --rm --net=host \
+        {{.AWSCliImage.RepoWithTag}} \
+        aws autoscaling describe-auto-scaling-groups \
+        --auto-scaling-group-name $AUTOSCALINGGROUP --region {{.Region}} \
+        --query 'AutoScalingGroups[].LaunchConfigurationName' --output text)"
+      {{end -}}
+
+      label() {
+        /usr/bin/docker run --rm -t --net=host \
+          -v /etc/kubernetes:/etc/kubernetes \
+          -v /etc/resolv.conf:/etc/resolv.conf \
+          -e INSTANCE_ID=${INSTANCE_ID} \
+          -e SECURITY_GROUPS=${SECURITY_GROUPS} \
+          {{if not .SpotFleet.Enabled -}}
+          -e AUTOSCALINGGROUP=${AUTOSCALINGGROUP} \
+          -e LAUNCHCONFIGURATION=${LAUNCHCONFIGURATION} \
+          {{end -}}
```

Comment:
I've verified this conditional to work as expected by adding both (1) ASG-based node pool and (2) Spot-fleet bsed node pool to my cluster and seeing node labels by running `kubectl get no --show-labels`

```
$ k get no --show-labels
NAME                                            STATUS    AGE       VERSION           LABELS
ip-10-0-0-158.ap-northeast-1.compute.internal   Ready     2h        v1.8.2+coreos.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/instance-type=c4.xlarge,beta.kubernetes.io/os=linux,failure-domain.beta.kubernetes.io/region=ap-northeast-1,failure-domain.beta.kubernetes.io/zone=ap-northeast-1a,kubernetes.io/hostname=ip-10-0-0-158.ap-northeast-1.compute.internal
ip-10-0-0-204.ap-northeast-1.compute.internal   Ready     2h        v1.8.2+coreos.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/instance-type=c4.xlarge,beta.kubernetes.io/os=linux,failure-domain.beta.kubernetes.io/region=ap-northeast-1,failure-domain.beta.kubernetes.io/zone=ap-northeast-1a,kube-aws.coreos.com/autoscalinggroup=k8s3-Controlplane-PUJZJTDHM2K2-Controllers-MKJ77KPR445K,kube-aws.coreos.com/cluster-autoscaler-supported=true,kube-aws.coreos.com/launchconfiguration=k8s3-Controlplane-PUJZJTDHM2K2-ControllersLC-HL1P0TRWX9Q8,kubernetes.io/hostname=ip-10-0-0-204.ap-northeast-1.compute.internal,node-role.kubernetes.io/master=
ip-10-0-0-79.ap-northeast-1.compute.internal    Ready     2h        v1.8.2+coreos.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/instance-type=c4.xlarge,beta.kubernetes.io/os=linux,failure-domain.beta.kubernetes.io/region=ap-northeast-1,failure-domain.beta.kubernetes.io/zone=ap-northeast-1a,kube-aws.coreos.com/autoscalinggroup=k8s3-Nodepool1-1H3X7UFLVK1P-Workers-19I4CJYIWPJT0,kube-aws.coreos.com/launchconfiguration=k8s3-Nodepool1-1H3X7UFLVK1P-WorkersLC-17165DG6TQG5R,kubernetes.io/hostname=ip-10-0-0-79.ap-northeast-1.compute.internal
```

As you see in `ip-10-0-0-158.ap-northeast-1.compute.internal`, for the node coming from the spot fleet we see no `autoscalinggroup` and `launchconfiguration` labels.
However it passes cfn-signal.service hence kube-node-label.service without errors.